addons/mail/static/tests/chatter/web/chatter.test.js

```diff
@@ -630,6 +631,53 @@ test("schedule activities on draft record should prompt with scheduling an activ
     await waitForSteps(["mail.activity.schedule"]);
 });
 
+test("schedule an activity on record should refresh parent view when hasParentReloadOnActivityChanged is true", async () => {
+    const wizardOpened = new Deferred();
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Test" });
+    mockService("action", {
+        doAction(action, options) {
+            if (action.res_model === "res.partner") {
+                return super.doAction(...arguments);
+            } else if (action.res_model === "mail.activity.schedule") {
+                asyncStep("mail.activity.schedule");
+                expect(action.context.active_model).toBe("res.partner");
+                expect(Number(action.context.active_id)).toBeGreaterThan(0);
+                options.onClose();
+                wizardOpened.resolve();
+            } else {
+                asyncStep("Unexpected action" + action.res_model);
+            }
+        },
+    });
+    let rpcCounter = 0;
+
+    onRpc("web_read", async (kwargs) => {
+        if (kwargs.model === "res.partner") {
+            rpcCounter++;
+        }
+    });
+
+    await start();
+    await openFormView("res.partner", partnerId, {
+        arch: `
+            <form>
+                <sheet>
+                    <field name="name"/>
+                </sheet>
+                <chatter reload_on_activity="True"/>
+            </form>`,
+    });
+    // A first load of the model should be done during the open form view
+    expect(rpcCounter).toBe(1);
+    await click("button", { text: "Activity" });
+    await wizardOpened;
+    await waitForSteps(["mail.activity.schedule"]);
+    await animationFrame();
+    // A second load of the model should be done after adding an activity
+    expect(rpcCounter).toBe(2);
```

Comment:
Manual deferred is redundant with asyncStep.

In any case, it is best whenever possible to wait changes in DOM rather than playing with timings.